### PR TITLE
Admin Attendance

### PR DESCRIPTION
**Pull request by RiyaSantore04**

### CODE REVIEW

File: app/controllers/admin_main/attendences_controller.rb [hunk 1]
```diff
@@ -0,0 +1,54 @@
...
+      redirect_to root_path, { notice: 'NOT FOUND :)' } unless @attendence.present?
+    end
+
+    def attendence_params
+      params.require(:attendence).permit(:checkout_time)
+    end
+
+    def employee_attendence(employee_id)
+      @attendences = Attendence.where(employee_id: employee_id)
+    end
```
Comment (chiraggargsass) [inline app/controllers/admin_main/attendences_controller.rb:47]: method define but never use

File: spec/controllers/admin_main/attendence_controller_spec.rb [hunk 2]
```diff
@@ -0,0 +1,62 @@
...
+
+  describe 'index' do
+    it 'Get Employee' do
+      get :index
+      expect(response.status).to eq(200)
+    end
+
+    it 'search employee' do
+      get :index, params: { employee: employee.name }
+      expect(response.status).to eq(200)
```
Comment (rorpyll) [inline spec/controllers/admin_main/attendence_controller_spec.rb:24]: expect with object

File: spec/controllers/admin_main/attendence_controller_spec.rb [hunk 3]
```diff
@@ -0,0 +1,62 @@
...
+    end
+  end
+
+  describe '#show' do
+    it 'show customer' do
+      get :show, params: { id: employee.id }
+      expect(response.status).to eq(200)
```
Comment (rorpyll) [inline spec/controllers/admin_main/attendence_controller_spec.rb:31]: expect with object

File: app/views/admin_main/employees/index.html.erb [hunk 4]
```diff
@@ -364,21 +364,31 @@
...
+                </td>
+                <td>
+                  <% if current_employee.is_admin? %>
+                    <%= link_to 'Attendence', admin_main_attendence_path(employee), class: "btn btn-primary"%> 
+                  <% end %>
+                </td>
+                <td>
+                  <% if current_employee.is_admin? %>
+                    <%= link_to 'Attendence', admin_main_attendence_path(employee), class: "btn btn-primary"%> 
+                  <% end %>
```
Comment (chiraggargsass) [inline app/views/admin_main/employees/index.html.erb:385]: why use same condition and button

Comment (RiyaSantore04) [reply app/views/admin_main/employees/index.html.erb:385]: sry, did by mistake

File: app/controllers/admin_main/attendences_controller.rb (hunk 1 above)
Comment (RiyaSantore04) [reply app/controllers/admin_main/attendences_controller.rb:47]: changes done

File: spec/controllers/admin_main/attendence_controller_spec.rb (hunk 3 above)
Comment (RiyaSantore04) [reply spec/controllers/admin_main/attendence_controller_spec.rb:31]: changes done

File: spec/controllers/admin_main/attendence_controller_spec.rb (hunk 2 above)
Comment (RiyaSantore04) [reply spec/controllers/admin_main/attendence_controller_spec.rb:24]: changes done

File: spec/controllers/admin_main/attendence_controller_spec.rb [hunk 5]
```diff
@@ -0,0 +1,62 @@
...
+  let!(:attendence) { FactoryBot.create(:attendence, employee_id: employee.id, checkin_time: Time.now) }
+
+  before do
+    sign_in(employee)
+  end
+
+  describe 'index' do
+    it 'Get Employee' do
+      get :index
+      expect(response.status).to eq(200)
```
[the remark this answers is not in the document]
Comment (RiyaSantore04) [reply spec/controllers/admin_main/attendence_controller_spec.rb:19]: changes done

File: config/routes.rb [hunk 6]
```diff
@@ -46,6 +46,7 @@
       resources :comments, only: [:create]
     end
     resources :employees
+    resources :attendences
```
[the remark this answers is not in the document]
Comment (RiyaSantore04) [reply config/routes.rb:49]: changes done

File: app/helpers/attendences_helper.rb [hunk 7]
```diff
@@ -1,2 +1,54 @@
...
+		unless hours.include?(nil)
+			hours.sum do |s|
+  				h, m = s.split(':').map(&:to_i)
+  				 60*h + m 
+			end.divmod(60).join(':')
+		end
+	end
+	def checkin_checkout_time(at)
+		time = []
+		cit = at.pluck(:checkin_time)
```
[the remark this answers is not in the document]
Comment (RiyaSantore04) [reply app/helpers/attendences_helper.rb:38]: changes done

File: app/models/attendence.rb [hunk 8]
```diff
@@ -1,35 +1,46 @@
...
-
-    def working_hour
-        unless self.checkout_time.nil?
-            data = (self.checkout_time - self.checkin_time).to_i
-            hour = (data/3600).to_i
-            min = ((data-(hour*3600)) / 60).to_i
-            sec = (data-(hour*3600) - (min*60)).to_i
-            self.hour = "#{hour}:#{min}"
-        end
+  self.inheritance_column = 'not_sti'
```
Comment (RiyaSantore04) [inline app/models/attendence.rb:4]: why

**Pull request merged**